docs/api-rest-docs.md

```diff
@@ -0,0 +1,281 @@
+
++ URL Parameters
+    + `source` (required) ... The data source (see "Source Permissions" above) to retrieve.
+
+Retrieve all image metadata (including URLs that can be used to retrieve file content) for a given `:source`.  You must
+have permission to view records associated with `:source` (see "Source Permissions" above).  If the HTTP `Accept` header
+is set to `application/json`, a JSON record will be returned.  Otherwise, a rendered HTML page will be returned.  If the
+user is logged in and has permission to write to `:source`, this page will contain controls to add or delete images.
+
+# `GET /api/images/:source/:uid`
+
++ HTTP Headers
+    + `Accept` ... What type of content (`application/json` or `text/html`) to return.
+
++ URL Parameters
+    + `uid` (required) ... The unique ID of a Unified Listing entry.
+    + `source` (required) ... The data source (see "Source Permissions" above) to retrieve.
+
+Retrieve all image metadata (including URLs that can be used to retrieve file content) for `:source` and `:uid`.  You
+must have permission to view records associated with `:source` (see "Source Permissions" above).  If the HTTP `Accept`
+header is set to `application/json`, a JSON record will be returned.  Otherwise, a rendered HTML page will be returned.
+If the user is logged in, this page will include controls to contribute metadata and new images.  If the user has
+permission to write to `:source`, contributions will be made to `:source` directly.  Otherwise, the user will save
+contributions to their own data source, e. g. `~username`.
+
+# `GET /api/images/file/:uid/:source/:image_id{?height&width}`
+
+Retrieve a single image file.  You must have permission to view records associated with `:source` (see "Source Permissions" above).
+
++ URL Parameters
+    + `uid` (required) ... The unique ID of the Unified Listing entry associated with this image.
+    + `source` (required) ... The data source (see "Source Permissions" above) to associate this image with.
+    + `image_id` (required) .. The ID of this image, as returned by `GET /api/images/file/gallery` (see below) or `POST /api/images/file/:uid` (see above).
+
++ Query Parameters
+    + `height` (optional, number) ... The desired height of the image.  If `height` is supplied and ``width` is omitted, the width will be determined using the aspect ratio of the original image.
+    + `width` (optional, number) ... The desired width of the image.  If `width` is supplied and `height` is omitted, the height will be determined using the aspect ratio of the original image.
+
+If both `height` and `width` are specified, the image will be scaled proportionately and mounted on a background with that
+exact size.  See [the Sharp documentation](http://sharp.dimens.io/en/stable/api/#embed) for details.
+
+# `HEAD /api/images/file/:uid/:source/:image_id{?height&width}`
+
+Returns just the HTTP request headers for a given image, including:
```

Comment:
Thanks - I think this info could helpfully go into a "Task" JIRA issue named something like "Implement strategy for reporting staleness for image resources" or so